src/libradosstriper/RadosStriperImpl.cc

```diff
@@ -1188,7 +1188,7 @@ int libradosstriper::RadosStriperImpl::aio_write_in_open_object(const std::strin
   // we need 3 references as striper_write_aio_req_complete will release two and
   // striper_write_aio_req_safe will release one
   auto cdata = ceph::make_ref<WriteCompletionData>(this, soid, lockCookie, c);
-  m_ioCtxImpl->get();
+  // m_ioCtxImpl->get();
```

Comment:
pls remove the line altogether

Reply:
The more important question:
the comment above explains why 3 refs are required. And the bug is because only 2 refs are freed.
So - if the comment is still correct, the get() should remain, and there should be a missing release to add somewhere. And if the comment is no longer accurate - please change it too.

Now - I'm not familiar with the code to know which of the two options is the right one. I'll try to pull others in.

Reply:
> pls remove the line altogether

done

Reply:
> The more important question: the comment above explains why 3 refs are required. And the bug is because only 2 refs are freed. So - if the comment is still correct, the get() should remain, and there should be a missing release to add somewhere. And if the comment is no longer accurate - please change it too.
> 
> Now - I'm not familiar with the code to know which of the two options is the right one. I'll try to pull others in.

from my understanding of the code, 3 refs is the reference count of `cdata`, and has nothing to do with `m_ioCtxImpl`. 
```c++
  1、librados::AioCompletion *unlock_completion =
    librados::Rados::aio_create_completion(cdata, rados_write_aio_unlock_complete, 0);
  2、nc->set_complete_callback(cdata, striper_write_aio_req_complete);
  3、nc->set_safe_callback(cdata, striper_write_aio_req_safe);
```

I suspect the reason for holding a reference to `m_ioCtxImpl` by calling `m_ioCtxImpl->get()` is to prevent m_ioCtxImpl from being released by other threads. However, the reference count should be released after the asynchronous write is completed.

Reply:
@dovefi Did you try to reproduce an issue on the main branch? I have tried to run the test case as you mentioned in the tracker bug, i am facing different issue not facing leak issue. I don;t think the issue will resolve after avoid the refcnt for m_ioCtxImpl in the function.

Reply:
> @dovefi Did you try to reproduce an issue on the main branch? I have tried to run the test case as you mentioned in the tracker bug, i am facing different issue not facing leak issue. I don;t think the issue will resolve after avoid the refcnt for m_ioCtxImpl in the function.

I have conducted stress testing on version 14.2.21. This PR indeed resolves the memory leak issue in the client application, but testing has not been done on the latest version.